pkg/apis/extensions/types.go

```diff
@@ -226,6 +226,14 @@ type DeploymentSpec struct {
 	// Value of this key is hash of DeploymentSpec.PodTemplateSpec.
 	// No label is added if this is set to empty string.
 	UniqueLabelKey string `json:"uniqueLabelKey,omitempty"`
+
+	// The config this deployment is rolling back to. Will be cleared after rollback is done.
+	Rollback *RollbackConfig `json:"rollback,omitempty"`
```

Comment:
If we are going to save info about the rollback in the spec and after @ghodss [comment](https://github.com/kubernetes/kubernetes/pull/19686#issuecomment-173061372) I may be convinced of saving the latest successfull* podtemplate as the rollback config. 

*It wouldn't make sense to save a template as the rollback template if it was never successfully deployed, right?

cc: @ironcladlou @smarterclayton